### PR TITLE
add Timex.Ecto.TimestampWithTimezone to support :timestamptz and deprecate Timex.Ecto.Timestamps

### PR DESCRIPTION
**Pull request by bitgorust**
@bitwalker , thanks for your job!

I have met two problems when using these excellent libraries. Here are the problems and my solutions. I hope I am not wasting your time.

#### Timex.Ecto.TimestampWithTimezone

Since a new `:datetimetz` type is needed when using `Timex.Ecto.DateTimeWithTimezone`, which is a pain to me, I have created `Timex.Ecto.TimestampWithTimezone` to directly support `:timestamptz` in my project and it works well. 

#### Timex.Ecto.Timestamps

I think `Timex.Ecto.Timestamps` which restricts `inserted_at` and `updated_at` to `Timex.Ecto.DateTime` is unnecessary as `timestamps` has all the options needed and is more flexible and convenient. Here's an example in my project:

```elixir
timestamps(type: Timex.Ecto.TimestampWithTimezone,
           autogenerate: {Timex.Ecto.TimestampWithTimezone, :autogenerate, [:usec]})
```

Looking forward to your response. And if my job is acceptable, I am willing to work further such as adding relative test cases and documents.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -29,7 +29,7 @@ defmodule TimexEcto.Mixfile do
   # Type `mix help deps` for more examples and options
   defp deps do
     [{:timex, "~> 3.0"},
-     {:ecto, "~> 2.1.0"},
+     {:ecto, git: "https://github.com/kenuyx/ecto.git", branch: "master"},
```
Comment (bitwalker) [inline mix.exs]: Can you remove this from the PR?

**Comment by bitwalker** [comment]
This looks great! Can you leave `Timex.Ecto.Timestamps` in the project, but update any references to it in the documentation to mention that it is deprecated in favor of the approach you've laid out here? I want to give folks a deprecation cycle before removing something that is likely used in many places.

File: mix.exs (hunk 1 above)
Comment (bitgorust) [reply mix.exs]: Oh of course, I am so careless.

**Comment by bitgorust** [comment]
Thanks for your approval. I will have a try tomorrow.

**Comment by bitgorust** [comment]
@bitwalker `lib/model.ex` is back and `README.md` is updated. A unit test for `Timex.Ecto.TimestampWithTimezone` is also added. Please check if I have done all these right.

File: mix.exs [hunk 2]
```diff
@@ -28,8 +28,8 @@ defmodule TimexEcto.Mixfile do
   #
   # Type `mix help deps` for more examples and options
   defp deps do
-    [{:timex, "~> 3.0"},
-     {:ecto, "~> 2.1.0"},
+    [{:timex, "~> 3.1"},
```
Comment (bitgorust) [inline mix.exs:31]: @bitwalker Version here seems to be at least 3.1.15, in which `to_datetime({{y,m,d},{h,mm,s,_us}} = dt, timezone)` was supported the first time.

**Review by bitwalker — APPROVED** [review]

**Comment by bitwalker** [comment]
Thanks! This will be published in 3.2.0

**Pull request merged**